src/ServiceControl/Operations/BodyStorage/BodyStorageFeature.cs

```diff
@@ -58,36 +58,39 @@ static string GetContentType(IReadOnlyDictionary<string, string> headers, string
                 return contentType;
             }
 
-            async Task StoreBody(byte[] body, IReadOnlyDictionary<string, string> headers, IDictionary<string, object> metadata, int bodySize, string contentType)
+            async ValueTask StoreBody(byte[] body, IReadOnlyDictionary<string, string> headers, IDictionary<string, object> metadata, int bodySize, string contentType)
             {
                 var bodyId = headers.MessageId();
+                var bodyUrl = string.Format(BodyUrlFormatString, bodyId);
                 var isBinary = contentType.Contains("binary");
                 var avoidsLargeObjectHeap = bodySize < LargeObjectHeapThreshold;
 
-                var bodyUrl = await StoreBodyInBodyStorage(body, bodyId, contentType, bodySize)
```

Comment:
I saw this last weekend and wondering why this was happening. Based on the changes is seems "dual" storing was a bug?

Reply:
I wasn't a bug but it was suboptimal. We store the body in each place for different reasons. The attachment storage was there from the beginning for every message because that gave us a binary blob we could use to recreate the message when returning to the source queue. The in-document storage is there purely to support the full text indexing and was never really intended to be used for retries.